app/controllers/concerns/images_utility.rb

```diff
@@ -0,0 +1,38 @@
+module ImagesUtility
+  extend ActiveSupport::Concern
+
+  private
+
+  def download_or_show_image(type, render_to, width)
+    html = image_html(type) 
+    render html: html.html_safe and return unless render_to == 'jpg' 
+    kit = build_kit(html, "#{type.tr('_', '-')}.css", width) 
+    send_data(kit.to_jpg, type: 'image/jpg', filename: "#{type}.jpeg")
+  end
+
+  def image_html(image_type)
+    render_to_string(partial: image_type,
+                     locals: { app_config: @app_configuration, user: @user,
+                               render_to: params[:render_to]&.to_sym,
+                               context: params[:context]&.to_sym,
+                               company: Company.find_by_id(params[:company_id]) })
```

Comment:
To make this cleaner we should not rely on instance vars being set in the controller, so we should pass in the arguments `app_config` and `user` to this method.   `user` could be a keyword argument that defaults to `nil` (since it is not need for the h-brand image).

While we're at it, we can add a `company` argument as well, which could also default to `nil` (since it is not needed for the proof-of-membership image).

https://robots.thoughtbot.com/ruby-2-keyword-arguments